2020-02-07  Nicholas Anthony Mark  <markn@lnxsrv07.seas.ucla.edu>

	* timespec.h: timespec_cmp
	Deletes subtraction implementation, since it is vulnerable to integer overflow
	which can lead to inaccurate comparisons. Instead, uses direct comparison, and
	if statements to determine what signed integer to return.


--- ./timespec.h	2020-02-07 01:04:25.823630000 -0800
+++ ../coreutils_install/timespec.h	2020-02-07 01:07:42.042311000 -0800
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec)
+    return 1;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
